// @typescript-eslint/scope-manager/dist/lib/esnext.bigint.js
"use strict";
// THIS CODE WAS AUTOMATICALLY GENERATED
// DO NOT EDIT THIS CODE BY HAND
// YOU CAN REGENERATE IT USING yarn generate:lib
Object.defineProperty(exports, "__esModule", { value: true });
exports.esnext_bigint = void 0;
exports.esnext_bigint = {
    BigIntToLocaleStringOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'BigIntToLocaleStringOptions',
        writeable: false,
    },
    BigInt: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'BigInt',
        writeable: false,
    },
    BigIntConstructor: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'BigIntConstructor',
        writeable: false,
    },
    BigInt64Array: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'BigInt64Array',
        writeable: false,
    },
    BigInt64ArrayConstructor: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'BigInt64ArrayConstructor',
        writeable: false,
    },
    BigUint64Array: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'BigUint64Array',
        writeable: false,
    },
    BigUint64ArrayConstructor: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'BigUint64ArrayConstructor',
        writeable: false,
    },
    DataView: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'DataView',
        writeable: false,
    },
    Intl: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: true,
        name: 'Intl',
        writeable: false,
    },
};
